src/common.c

```diff
@@ -538,12 +539,69 @@ static int ofi_str_to_sin6(const char *str, void **addr, size_t *len)
 	return 0;
 }
 
+static int ofi_hostname_toaddr(const char *name, uint32_t *addr_format,
+			       void **addr, size_t *len)
+{
+	struct addrinfo *ai;
+	int err;
+
+	err = getaddrinfo(name, NULL, NULL, &ai);
+	if (err)
+		return err;
+
+	*addr_format = (ai->ai_family == AF_INET6) ? FI_SOCKADDR_IN6 : FI_SOCKADDR_IN;
+	*len = ai->ai_addrlen;
+	*addr = calloc(1, *len);
+	if (!*addr)
+		return -FI_ENOMEM;
+
+	memcpy(*addr, ai->ai_addr, *len);
+
+	freeaddrinfo(ai);
+	return 0;
+}
+
+static int ofi_ifname_toaddr(const char *name, uint32_t *addr_format,
+			     void **addr, size_t *len)
+{
+        struct ifreq ifr;
+        int sock_fd, err;
+
+        sock_fd = socket(PF_INET, SOCK_STREAM, 0);
+	if (sock_fd < 0)
+		return sock_fd;
+
+        snprintf(ifr.ifr_name, IFNAMSIZ, "%s", name);
+        err = ioctl(sock_fd, SIOCGIFADDR, &ifr);
+	if (err)
+		goto out;
+
+	*addr_format = FI_SOCKADDR;
+	*len = sizeof(struct sockaddr);
+	*addr = calloc(1, *len);
+	if (!*addr) {
+		err = -FI_ENOMEM;
+		goto out;
+	}
+
+	memcpy(*addr, &ifr.ifr_addr, *len);
+
+out:
+	close(sock_fd);
+	return err;
+}
```

Comment:
There's existing code, ofi_getifaddrs(), that's OS portable.  Can this be used instead?

Reply:
That can be used. The catch is that getifaddrs is not always present (it's guarded by a configure generated definition). I think I can use ofi_getifaddrs whenever possible and use the current code as a fallback.

Reply:
It's guarded mainly because the calls are not available on windows.  See the appveyor failures.